cmake/autocoder/deployment_validation.cmake

```diff
@@ -0,0 +1,42 @@
+####
+# autocoder/deployment_validation.cmake
+#
+# An autocoder that performs validation steps on the deployment to ensure that it is well-formed. This will produce
+# warning output when validation fails.
+#####
+include(utilities)
+include(autocoder/helpers)
+
+autocoder_setup_for_multiple_sources()
+
+####
+# `deployment_validation_is_supported`:
+#
+# Required function, processes .fpp files
+####
+function(deployment_validation_is_supported AC_INPUT_FILE)
+    autocoder_support_by_suffix(".fpp" "${AC_INPUT_FILE}") # No re-scan,  already done in FPP
+endfunction (deployment_validation_is_supported)
+
+####
+# `deployment_validation_is_supported`:
+#
+# Required function, look for .fpp files defining a topology block and ensure the following equivalence:
+#     ${PROJECT_NAME} == topology name
+####
+function(deployment_validation_setup_autocode AC_INPUT_FILES)
+    foreach(AC_FILE IN LISTS AC_INPUT_FILES)
+        file(READ "${AC_FILE}" FILE_TEXT)
+        # Does this file match the pattern (module { topology {} })
+        if (NOT FPRIME_SKIP_PROJECT_NAME_VALIDATION AND
+            FILE_TEXT MATCHES "\n?[^\n#@]* topology ([a-zA-Z0-9_]+)" AND
```

Comment:
Is the intent here **not** to match commented out topologies? I don't think this is what's happening since the leading newline is optional. I confirmed that adding a 
```# topology Refff {}``` somewhere in an FPP file will get caught and throw a warning.  Although, as long as it catches the actual topology, it's not that big a deal if it throws false positives.

Reply:
Good catch!

I'd like to avoid false positives. I'll attempt resolution.